spark-bigquery-tests/src/main/java/com/google/cloud/spark/bigquery/integration/QueryPushdownIntegrationTestBase.java

```diff
@@ -468,4 +431,501 @@ public void testAggregateExpressions() {
     assertThat(r1.get(10)).isEqualTo(0.25); // VAR_POP(num1)
     assertThat(r1.get(11)).isEqualTo(0.5); // VAR_SAMP(num1)
   }
+
+  @Test
+  public void testInnerJoin() {
+    writeTestDataToBigQuery(
+        getNumStructDataFrame(TestConstants.numStructDataset),
+        testDataset.toString() + "." + testTable);
+    Dataset<Row> df = readTestDataFromBigQuery(testDataset.toString() + "." + testTable);
+
+    // Creating a DataFrame of schema NumStruct, and writing it to BigQuery
+    writeTestDataToBigQuery(
+        getNumStructDataFrame(TestConstants.numStructDatasetForJoin),
+        testDataset.toString() + "." + testTable + "_to_join");
+    // Creating an additional DataFrame of schema NumStruct, and writing it to BigQuery which will
```

Comment:
Looks like the comments are misplaced :)

Reply:
good catch. will update it 😬